go/scheduler/tasks.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+
+from go.vumitools.tests.helpers import djangotest_imports
+
+with djangotest_imports(globals()):
+    from celery.task import task, group
+
+    from go.scheduler.models import PendingTask, Task
+    from go.base.utils import (
+        get_conversation_view_definition, vumi_api)
+
+
+@task()
+def perform_task(pending_id):
+    """ Perform a task. """
+    pending = PendingTask.objects.get(id=pending_id)
+    if pending.started_timestamp is None:
+        pending.started_timestamp = datetime.datetime.utcnow()
+        pending.save()
+    else:
+        return
```

Comment:
What if we moved the `started_timestamp` onto the `Task` model instead of the `PendingTask` model? Then we would retain the information on when the task was run even after the pending task was deleted.

We probably need to handle the `pending_id` no longer existing either way (and treat it as if the task had already run).

We should probably also log a message saying that the task had already been executed so we can see what happened (using celery's task logging helper).